### PR TITLE
Tokenomics - APR

### PR DESCRIPTION
**Pull request by 1saf**
# Description

Brings in the APR calculations for staked pools

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202055021734852

### CODE REVIEW

**Review by garethfuller — COMMENTED** [review]
@1saf There seems to be a lot of changes here that are unrelated to the APR change?

**Review by TomAFrench — COMMENTED** [review]
Things look ok other than a magical factor of 16.48 cropping up. I think we might be overcomplicating things by building so many maps of intermediate steps.

I'd separate the maths for calculating the APR for a pool out into a separate function and then iterate over the gauges once rather than 3+ times as we're doing here. This would be easier to parse.

File: src/services/staking/utils.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    relativeWeights,
+    gaugeAddress,
+    workingSupply: workingSupply.toString(),
+    relativeWeight: relativeWeight.toString(),
+    balPayable: balPayable.toString(),
+    weeklyReward: weeklyReward.toString(),
+    yearlyReward: yearlyReward.toString(),
+    inflationRate: inflationRate.toString(),
+    bptPrice: bptPrice.toString()
+  });
```
Comment (garethfuller) [inline src/services/staking/utils.ts:69]: Remove before merge

**Review by garethfuller — CHANGES_REQUESTED** [review]
Looks pretty good. Main thing is that for ranges displayed it should include all APR components, not just the staking APR.

A couple of things missing
- Totals in APR tooltips should be a range and include the staking APRs
- APR on the pool page stat card should be a range and include the staking APRs
- Invest flow weekly yield should be a range using total APR range (maybe I can help with this part tomorrow?)

Screenshot 2022-04-05 at 20 22 43
Screenshot 2022-04-05 at 20 24 53

File: src/composables/useNumbers.ts [hunk 2]
```diff
@@ -40,7 +46,7 @@ enum PresetFormats {
   usd_m = '$0,0.00a',
   percent = '0.00%',
   percent_variable = '0.[0000]%',
-  percent_lg = '0%'
+  percent_lg = '0,0%'
```
[the remark this answers is not in the document]
Comment (1saf) [reply src/composables/useNumbers.ts:49]: doesnt work with fnum2

File: src/lib/utils/index.ts [hunk 3]
```diff
@@ -137,3 +139,7 @@ export function formatWordListAsSentence(
 export function getAddressFromPoolId(poolId: string) {
   return poolId.substring(0, 42);
 }
+
+export function getBalAddress() {
+  return TOKENS.AddressMap[configService.network.chainId]?.BAL;
+}
```
[the remark this answers is not in the document]
Comment (1saf) [reply src/lib/utils/index.ts:145]: will do

File: src/pages/index.vue [hunk 4]
```diff
@@ -98,6 +98,7 @@ function navigateToCreatePool() {
 
 <template>
   <div class="lg:container lg:mx-auto pt-10 md:pt-12">
+    {{ gaugeAprMap }}
```
[the remark this answers is not in the document]
Comment (1saf) [reply src/pages/index.vue:101]: to remove

File: src/services/balancer/contracts/contracts/token-admin.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    private readonly config = configService,
+    private readonly web3 = web3Service
+  ) {
+    this.instance = new Contract(this.address, this.abi, this.provider);
+  }
+
+  async getInflationRate() {
+    const rate = await this.instance.getInflationRate();
+    return formatUnits(rate, 18);
+  }
```
[the remark this answers is not in the document]
Comment (1saf) [reply src/services/balancer/contracts/contracts/token-admin.ts:26]: yeah something to visit

**Pull request merged**